mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -378,20 +378,20 @@ LogicalResult ForOp::verifyRegions() {
   return success();
 }
 
-std::optional<Value> ForOp::getSingleInductionVar() {
-  return getInductionVar();
+std::optional<SmallVector<Value>> ForOp::getLoopInductionVars() {
+  return SmallVector<Value, 1>{getInductionVar()};
```

Comment:
I suspect using different number of stack-based elements in small vector between this and the return type will prevent move semantics here.